pkgs/applications/editors/emacs/sources.nix

```diff
@@ -4,8 +4,8 @@
 }:
 
 let
-  mkArgs = { pname, version, variant, patches ? _: [ ], rev, hash }: {
-    inherit pname version variant patches;
+  mkArgs = { pname, version, variant, templateNativeCompDriverOptionsPatch, patches ? _: [ ], rev, hash }: {
+    inherit pname version variant templateNativeCompDriverOptionsPatch patches;
```

Comment:
This specific patch changes almost every release so I agree with the idea of moving it to `mkArgs` from `make-emacs.nix`.  However, not putting it in `patches` seems a bit over-engineering to me.  Just my 2 cents.

Reply:
Well, give than we are passing the same patch to both the same version of `emacs` and `emacs-macport`, now I think it is better to not moving the patch into `mkArgs` from `make-emacs.nix`.

Reply:
Why not?

The patch depends on the Emacs source being built. It looks reasonable to parameterize it.
This is similar to the patches we are using for Emacs 28.

Reply:
> Why not?

To reduces code duplication.

---

Thinking it a bit more, I am fine with keeping it in `make-emacs.nix` or putting it into `patches` of `mkArgs`.